_specifications/specification-3-17.md

```diff
@@ -8895,6 +8915,197 @@ export interface Moniker {
+
+> *Since version 3.17.0*
+
+The inline values request is sent from the client to the server to compute inline values for a given text document that may be rendered in the editor at the end of lines.
+
+_Client Capability_:
+* property name (optional): `textDocument.inlineValues`
+* property type: `InlineValuesClientCapabilities` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValuesClientCapabilities" name="inlineValuesClientCapabilities" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlineValuesClientCapabilities {
+	/**
+	 * Whether inline values supports dynamic registration.
+	 */
+	dynamicRegistration?: boolean;
+}
+```
+
+_Server Capability_:
+* property name (optional): `inlineValuesProvider`
+* property type: `InlineValuesOptions` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValuesOptions" name="inlineValuesOptions" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlineValuesOptions extends WorkDoneProgressOptions {
+}
+```
+
+_Registration Options_: `InlineValuesRegistrationOptions` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValuesRegistrationOptions" name="inlineValuesRegistrationOptions" class="linkableAnchor"></a></div>
+
+```typescript
+export interface InlineValuesRegistrationOptions extends
+	TextDocumentRegistrationOptions, InlineValuesOptions {
+}
+```
+
+_Request_:
+* method: `textDocument/inlineValues`
+* params: `InlineValuesParams` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValuesParams" name="inlineValuesParams" class="linkableAnchor"></a></div>
+
+```typescript
+interface InlineValuesParams extends WorkDoneProgressParams {
+	/**
+	 * The document to request inline values for.
+	 */
+	readonly textDocument: TextDocumentIdentifier;
+
+	/**
+	 * The visible document range for which inline values should be computed.
+	 */
+	readonly viewPort: Range;
+
+	/**
+	 * Additional information about the context in which inline values were
+	 * requested.
+	 */
+	readonly context: InlineValuesContext;
+}
+```
+
+<div class="anchorHolder"><a href="#inlineValuesContext" name="inlineValuesContext" class="linkableAnchor"></a></div>
+
+```typescript
+interface InlineValuesContext {
+	/**
+	 * The document range where execution has stopped.
+	 * Typically the end position of the range denotes the line where the inline values are shown.
+	 */
+	readonly stoppedLocation: Range;
+}
+```
+
+_Response_:
+* result: `InlineValue[]` \| `null` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValue" name="inlineValue" class="linkableAnchor"></a></div>
+
+```typescript
+/**
+ * Inline value information can be provided by different means:
+ * - directly as a text value (class InlineValueText).
+ * - as a name to use for a variable lookup (class InlineValueVariableLookup)
+ * - as an evaluatable expression (class InlineValueEvaluatableExpression)
```

Comment:
How would the expression be evaluated? It seems like a lot of this api relies on an execution context like a debugger? I could see this working with DAP but seems out of place in LSP